File: 1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (seen.containsKey(diff)) {
                return new int[]{seen.get(diff), i};
            }
            seen.put(num, i);
        }
        return new int[]{};
    }
}

// Time complexity: O(n)
// Space complexity: O(n)

File: 1. Two Sum/solution.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> prev;
    
            for (int i = 0; i < nums.size(); i++) {
                int num = nums[i];
                int diff = target - num;
                if (prev.find(diff) != prev.end()) {
                    return {prev[diff], i};
                }
                prev[num] = i;
            }
    
            return {};
        }
};

// Time Complexity: O(n)
// Space Complexity: O(n)